selfservice/hook/web_hook_integration_test.go

```diff
@@ -899,36 +901,23 @@ func TestAsyncWebhook(t *testing.T) {
 	}
 	// at this point, a goroutine is in the middle of the call to our test handler and waiting for a response
 	incomingCancel() // simulate the incoming Kratos request having finished
+	close(blockHandlerOnExit)
 	timeout := time.After(200 * time.Millisecond)
-	for done := false; !done; {
-		if last := logHook.LastEntry(); last != nil {
-			msg, err := last.String()
-			require.NoError(t, err)
-			assert.Contains(t, msg, "Dispatching webhook")
+	var found bool
+	for !found {
+		for _, entry := range logHook.AllEntries() {
+			if entry.Message == "Webhook request succeeded" {
```

Comment:
Make this independent of ordering and just look whether we find the succeed hook :)